Add GitHub workflow to test deploy.sh automatically on push & PR

mmlakin opened the pull request:
This requires the ability to run deploy.sh without prompts, so add a flag "--bypass-yn-prompt" to deploy.sh which will cause the yn-prompt function to always return "yes" (value: 0).

mmlakin committed:
Add --bypass-yn-prompt flag to deploy.sh

Allows deploy.sh to run automatically without requiring any prompting.
Modifies yn_prompt function to always return 0 (yes) if flag is present.

Main driver for this functionality is enabling automated workflow testing in
GitHub.

mmlakin committed:
Add GitHub workflow to test deploy.sh

Creates GitHub workflow named "Test deploy.sh (Mac & Linux)" which
automatically tests running deploy.sh with the --bypass-yn-prompt flag
on both MacOS and Ubuntu, installing all required software and plugins.
This workflow is trigged automatically for every push and PR.

Test output can be viewed on GitHub via the repo's Actions tab.

braxeatssnacks commented on deploy.sh:
```diff
@@ -6,6 +6,9 @@ tmux_layouts_dir="$HOME/.tmuxinator"
 vim_config_dir="$HOME/.vim"
 vim_config_path="$HOME/.vimrc"
 zsh_config_path="$HOME/.zshrc"
+if [[ $1 = '--bypass-yn-prompt' ]]; then
```
This is awesome! However, in the event we decide to extend custom functionality and add other flags this won't be immediately extendable. 

Totally non-blocking, but might I suggest adapting this first parameter check into a loop which runs through all parsed command line args?

```
while [[ "$1" != "" ]]; do
    case $1 in
      --bypass-yn-prompt) shift
                          bypass_yn_prompt="${bypass_yn_prompt:-true}"
                          ;;
      *)				  exit 1   # invalid arg
    esac
    shift
  done
```

Note this extension `"${ENVIRONMENT_VARIABLE:-VALUE}"` is nice because it respects configuration that is passed in via environment variable 😄.

braxeatssnacks replied:
This could also be placed into the `main` function, since the scope of this variable is global if we felt it appropriate.

mmlakin replied:
This is cool, we should definitely do this when we build out the ability to customize the install and run without prompts in the future.  For now, I'm OK with keeping it with keeping it simple just to facilitate the automated testing, but it does make more sense to move the argument handling to the bottom of the script before main() is called instead of right at the top.
Also, since this is purely for automated testing ability, we could just use an environment variable check instead of a cmd line argument, but this works for now and is visible in the actions config.

braxeatssnacks reviewed (dismissed):
🚀👨

mmlakin committed:
Move arg handling below main()

It makes more sense to do arg handling below main() where the script
execution starts, rather than at the top where variables are defined.
